desc/objectives/utils.py

```diff
@@ -41,40 +54,50 @@ def factorize_linear_constraints(constraints, objective_args):  # noqa: C901
         and recovering x from y.
 
     """
+    for con in constraints:
+        for thing in con.things:
+            if thing not in objective.things:
+                warnings.warn(
+                    f"Optimizable object {thing} is constrained"
```

Comment:
Could be useful to specify which constraint is the culprit, like `f"Optimizable object {thing} is constrained by {con}"`

Reply:
It would also be really useful if we could print out what the free variables are when running an optimization (so the opposite of this). That way you can check to make sure you included all the constraints you meant to.

Reply:
@dpanici good idea. 
@ddudt that might get messy, since often there are many free variables. did you have anything particular in mind?